### PR TITLE
test: describe pod object in error scenario

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

Adds some additional pod introspection foo in the service of debugging #679. This will have general value, so "kubectl pod describe" has been added to the error gathering implementation for the various "wait for all pods to become ready" tests.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: test/e2e/kubernetes/pod/pod.go [hunk 1]
```diff
@@ -498,6 +498,10 @@ func WaitOnReady(podPrefix, namespace string, successesNeeded int, sleep, durati
 					if e != nil {
 						log.Printf("Unable to print pod logs for pod %s", p.Metadata.Name)
 					}
+					e = p.Describe()
+					if e != nil {
+						log.Printf("Unable to describe pod %s", p.Metadata.Name)
```
Comment (tariq1890) [inline test/e2e/kubernetes/pod/pod.go]: can we also print the error?

Comment (jackfrancis) [reply test/e2e/kubernetes/pod/pod.go]: done

File: test/e2e/kubernetes/pod/pod.go [hunk 2]
```diff
@@ -793,6 +797,17 @@ func (p *Pod) Logs() error {
 	return nil
 }
 
+// Describe will describe a pod resource
+func (p *Pod) Describe() error {
+	cmd := exec.Command("k", "describe", "pod", p.Metadata.Name, "-n", p.Metadata.Namespace)
+	out, err := util.RunAndLogCommand(cmd, commandTimeout)
+	log.Printf("\n%s\n", string(out))
+	if err != nil {
```
Comment (tariq1890) [inline test/e2e/kubernetes/pod/pod.go]: Lines 805-808 can be reduced to `return err`

**Pull request merged**